Source/JavaScriptCore/runtime/JSStringJoiner.h

```diff
@@ -77,6 +78,13 @@ inline JSStringJoiner::JSStringJoiner(JSGlobalObject* globalObject, LChar separa
         throwOutOfMemoryError(globalObject, scope);
 }
 
+inline JSValue JSStringJoiner::join(JSGlobalObject* globalObject)
+{
+    if (m_strings.size() == 1)
+        return jsString(globalObject->vm(), m_strings[0].view.toString());
```

Comment:
Why doesn't this use the underlying string if present? Seems that doing that could avoid allocating a new `StringImpl` in some cases. Maybe those aren't common enough.

Reply:
I assumed the string view might point to only part of the underlying string? I am not very familiar with this code.

Reply:
That’s correct. But:

1. if the lengths are equal, the underlying string can be used as is — I think this is a common case and the optimization of not allocating any memory for a `StringImpl` would be significant
2. if the string view’s length is smaller than the underlying string, we _could_ use `substringSharingImpl` to create a new `StringImpl` that is small and shares the characters, but that may be a more difficult tradeoff since we could keep a large string alive, and may not be as common a case

I was really thinking of (1) above, and if it’s common enough it might lead to a bigger performance win.

Reply:
Thanks for the clarification. I'll see if we can get a bigger win.

Reply:
You are right that (1) is the common case. 99.1% of the cases in Speedometer where m_strings has a length of 1 in fact. I'll do some A/B testing on the bots.

Reply:
Hmm, I tried the following patch (https://pastebin.com/Nwacq8mF) but A/B testing says this is no longer a progression. Weird.